### ISSUE TITLE
Page.addJavaScript() fails to load files from the `frontend/` folder with 404

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#8320]
I've created `frontend/src/test.js` file in the Skeleton Starter:

```
function test(val){
    alert('Hi'+val);
}
```

In MainView.java's constructor I'm calling the following function:

```
UI.getCurrent().page.addJavaScript("./src/test.js")
```

However, the `test.js` file is not loaded: the browser reports that http://localhost:8080/frontend/src/test.js is a 404 and the JavaScript console complains that

```
Loading failed for the <script> with source “http://localhost:8080/frontend/src/test.js”.
```

However, according to Page.addJavaScript() javadoc:

> Relative URLs are interpreted as relative to the configured {@code frontend} directory location.

Calling `UI.getCurrent().page.executeJs("test('foo')")` from Button click handler complains that the `test()` function is undefined.

Vaadin 14.1.27

**Comment by Legioth** [comment]
That Javadoc is outdated and only relevant for Bower mode.

With npm, the `/frontend` dir is used for anything that should be processed by webpack whereas regular static file locations (e.g `/src/main/webapp` or `/src/main/resources/META-INF/resources`) are for files that should be used as-is without processing.

`addJavaScript` is always using files as-is since knowledge about such files cannot even be known at compile time. For this reason, such files should not be in the `/frontend/` directory. The documentation should be updated accordingly.

**Comment by denis-anisimov** [comment]
@mvysny , are you aware of the fact that this is a javadoc only issue ?
Just to be sure that you don't expect anything else .........
To be able to use dynamic JS you should put your file into the static resources location as Leif mentioned.
So this is the correct way to use `UI.getCurrent().page.addJavaScript` (even though I wouldn't say that `UI.getCurrent()` is the correct way but that's offtopic).
I'm asking because this is a BFP.

**Comment by mvysny** [comment]
@denis-anisimov yes, I'm completely fine with that. I'm happy to place stuff into src/main/webapp. The documentation needs to be fixed though, since it's misleading.

**Comment by denis-anisimov** [comment]
Sure.
Alright, thanks.
